### PR TITLE
Json data

### PR DESCRIPTION
**Pull request by SokratisVidros**

### CODE REVIEW

File: lib/adapters/defaultAdapter.js [hunk 1]
```diff
@@ -1,89 +1,26 @@
...
+const constants = require('../constants');
+const transformOutboundRes = require('../transformers/transformOutboundRes');
+const transformInboundReq = require('../transformers/transformInboundReq');
+const transformOutboundReq = require('../transformers/transformOutboundReq');
+const transformInboundRes = require('../transformers/transformInboundRes');
+const transformError = require('../transformers/transformError');
 
 function requestProxy(options) {
-    const { logger, serialize, outboundRequestId } = options;
+    const { logger, outboundRequestId } = options;
```
Comment (SokratisVidros) [inline lib/adapters/defaultAdapter.js]: Please move destructuring to the signature

File: lib/adapters/defaultAdapter.js [hunk 2]
```diff
@@ -125,106 +56,20 @@ function onInboundRequest({ ctx }) {
...
-            log_tag: this.constructor.EVENT.OUTBOUND_RESPONSE
-        }
-    );
-    const msg = `[${data.requestId}] ${MSG.OUTBOUND_RESPONSE} ${ctx.logCtx.method} ${ctx.logCtx.path} ${data.status} \
-${data.duration}ms | ${this.serialize(data)}`;
-
-    isStatusError(status) ?
-        setImmediate(this.logger.error.bind(this.logger), msg) :
-        setImmediate(this.logger.info.bind(this.logger), msg);
+    const transformedRes = transformOutboundRes.call(this, { ctx, health: this.health });
```
Comment (SokratisVidros) [inline lib/adapters/defaultAdapter.js]: Please check whether this call is necessary

File: lib/adapters/defaultAdapter.js [hunk 3]
```diff
@@ -125,106 +56,20 @@ function onInboundRequest({ ctx }) {
...
-        log_tag: this.constructor.EVENT.UNEXPECTED_ERROR
-    };
-    err.context = Object.assign(err.context || {}, ctx.logCtx);
-
-    setImmediate(this.logger.info.bind(this.logger), err);
+    const transformedErr = transformError({ ctx, err });
+    setImmediate(this.logger.error.bind(this.logger), transformedErr);
 }
 
 module.exports = () => {
```
Comment (SokratisVidros) [inline lib/adapters/defaultAdapter.js]: Is this function wrapping necessary?

File: lib/constants.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        INBOUND_REQUEST: {
+            MSG: '<--',
+            TAG: 'inbound_request'
+        },
+        OUTBOUND_RESPONSE: {
+            MSG: '-->',
+            TAG: 'outbound_response'
+        },
+        OUTBOUND_REQUEST: {
+            MSG: '->',
```
Comment (nikostoulas) [inline lib/constants.js:12]: Maybe `~~>` and `<~~`?

File: lib/appenders/log4jsStdoutAppender.js [hunk 5]
```diff
@@ -18,9 +20,38 @@ function stdoutAppender(layout, options) {
...
+            }
+        }
+
+        if (!(loggingEvent.data[0] instanceof Error) &&
+                typeof loggingEvent.data[0] === 'object' &&
+                loggingEvent.data[0] !== undefined) {
+
+            const data = loggingEvent.data[0];
+            switch (data.log_tag) {
+                case constants.CATEGORY.OUTBOUND_RESPONSE.TAG:
```
Comment (SokratisVidros) [inline lib/appenders/log4jsStdoutAppender.js]: Please check whether this  can be refactored with a declarative API via config.json.

File: lib/transformers/isHealthEndpoint.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+module.exports = (ctx, healthConfig) => {
+    if (healthConfig) {
```
Comment (SokratisVidros) [inline lib/transformers/isHealthEndpoint.js]: Amazing! Let's document that since it's a feature.

File: lib/transformers/transformInboundRes.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const constants = require('../constants');
```
Comment (SokratisVidros) [inline lib/transformers/transformInboundRes.js]: Prefer a shorter name such as `inboundRes.js`.

File: lib/transformers/transformInboundReq.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    let metaBody;
+    if (method === 'POST' && bodyKeys) {
+        const picked = lodash.pick(ctx.request.body, bodyKeys);
+        if (Object.keys(picked).length) {
+            metaBody = flat({body: picked}, {safe: true});
+        }
+    }
+
+    // pick headers
+    let metaHeaders = {};
```
Comment (SokratisVidros) [inline lib/transformers/transformInboundReq.js]: Please move before pick body since it's more reasonable.

**Review by SokratisVidros — COMMENTED** [review]
Impressive work!

I am just noting some steps for next steps.
- Please comment the changes added to serializeToKeyValue.js file
- Add linter
- Add MIT license

File: lib/adapters/defaultAdapter.js [hunk 9]
```diff
@@ -1,89 +1,26 @@
 const lodash = require('lodash');
```
Comment (SokratisVidros) [inline lib/adapters/defaultAdapter.js]: Maybe a shorter name here, such as defaultJs?

**Pull request merged**